documentation/decisions/adr/2024-03-19-dashboard-storage.md

```diff
@@ -0,0 +1,87 @@
+# Dashboard Data Storage
+
+- **Status:** Active
+- **Last Modified:** 2024-03-19
+- **Related Issue:** [#{issue number}](https://github.com/HHS/simpler-grants-gov/issues/{issue number})
+- **Deciders:** Aaron Couch, Billy Daly
+
+## Context and Problem Statement
+
+We want to determine the data storage solution for our upcoming business intelligence dashboard project.
+
+This data storage solution will be our canonical source of truth for many types of imported data. These data types will range from thousands of records a month (ex. sprint data), up to potentially millions of records of records (ex. Google Analytics) once this project reaches maturity. The chosen solution should be easy to use, cost-effective, and performant for all of our in-scope data types. The types of data we will be importing include:
+
+- Google Analytics
+- Grants program data, eg.
+  - Grants opportunities from the database
+  - Grant applications from S3
+- USA spending data CSVs
+- Communication platform stats, eg.
+  - Slack stats
+  - Google Group stats
+  - GitHub stats
+- API application metrics
+- API infrastructure metrics from Cloudwatch
+
+We will not be importing all of these types of data immediately. On the 0 - 6 month timeframe, we will only be importing the smaller datasets (thousands of records). By 2 - 5 years we will be importing all of these types of data, and our data size will be quite large (many millions of records). The desired solutions have different cost/performance characteristics in those time ranges, and we will need to evaluate those differences.
```

Comment:
One small note on this: I imagine by the 6 month mark we _might_ be ingesting analytics and infrastructure metrics. We'd certainly want to start doing that by the 12-month mark.

I think the biggest outstanding question for those sources of data is whether we want to ingest point level data (e.g. individual page views, clicks, and API calls) or if we'd do some level of aggregation _before_ loading it into our data warehouse.

Just to "show our work" around volume of data metrics:

- ~250k users in the past 7 days (based on latest GA metrics for grants.gov)
- ~4 average page views per user per visit (conservative estimate for sizing purposes)
- 52 weeks per year
- ~52 million page view records per year

Reply:
This number could grow geometrically though if we also want to track things like impressions and clicks and tie those to sessions and active devices.